### PR TITLE
Nc refactor/remove default datasource config encryption

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nocodb/src/cli.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export { SqlClientFactory } from '~/db/sql-client/lib/SqlClientFactory';
+export { MetaTable } from '~/utils/globals';
+export * from '~/utils/encryptDecrypt';
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/cli.ts:3]: we want to expose all functions from that module

File: packages/nocodb/src/providers/init-meta-service.provider.ts [hunk 2]
```diff
@@ -96,6 +97,9 @@ export const InitMetaServiceProvider: FactoryProvider = {
     // load super admin user from env if env is set
     await initAdminFromEnv(metaService);
 
+    // encrypt datasource if secret is set
+    await initDataSourceEncryption(metaService);
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/providers/init-meta-service.provider.ts:101]: if encryption fails app won't work as expected

File: packages/nocodb/src/models/Integration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        andWhere: jest.fn().mockReturnThis(),
+        clone: jest.fn().mockReturnThis(),
+        limit: jest.fn().mockReturnThis(),
+        offset: jest.fn().mockReturnThis(),
+        orderBy: jest.fn().mockReturnThis(),
+        then: jest
+          .fn()
+          .mockImplementation((callback) =>
+            Promise.resolve(callback(mockIntegrations)),
+          ),
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/models/Integration.spec.ts:57]: It's not normal Promise so we need to add additional props as well

File: packages/nocodb/src/meta/migrations/v2/nc_063_encrypt_flag.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import type { Knex } from 'knex';
+import { MetaTable } from '~/utils/globals';
+
+const up = async (knex: Knex) => {
+  await knex.schema.alterTable(MetaTable.SOURCES, (table) => {
+    table.boolean('is_encrypted').defaultTo(false).index();
+  });
+  await knex.schema.alterTable(MetaTable.INTEGRATIONS, (table) => {
+    table.boolean('is_encrypted').defaultTo(false).index();
+  });
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nocodb/src/meta/migrations/v2/nc_063_encrypt_flag.ts]: There is an upgrader for decrypting all existing config.

File: packages/nc-secret-mgr/src/index.spec.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+describe('index.ts', () => {
+  it('should parse the arguments and options correctly', () => {
+    const argv = ['node', 'index.ts', 'prevSecret', 'newSecret', '--nc-db','test_db_url', '--database-url', 'test_db_url', '-o', 'prevSecret', '-n', 'newSecret'];
+    program.parse(argv);
+    expect(program.opts().prev).to.equal('prevSecret');
+    expect(program.opts().new).to.equal('newSecret');
+    expect(program.opts().ncDb).to.equal('test_db_url');
+    expect(program.opts().databaseUrl).to.equal('test_db_url');
+  });
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-secret-mgr/src/index.spec.ts:15]: Will handle in a followup PR

**Comment by pranavxc** [comment]
@coderabbitai pause

**Review by dstala — APPROVED** [review]

**Pull request merged**